### PR TITLE
Remove .NET Core 3 Dev channel

### PR DESCRIPTION
**Pull request by mmitche**
The .NET Core 3 Dev Channel from arcade and documentation. It is no longer in use and will be removed from BAR soon.

### CODE REVIEW

**Comment by riarenas** [comment]
Looks like all default channels and subscriptions are already gone, so this should be safe 😄

File: Documentation/DependencyFlowOnboarding.md [hunk 1]
```diff
@@ -186,7 +186,7 @@ How often does a source repo flows dependencies into a target repo.
 
 ##### 3.5. Create a channel (optional, typically not needed)
 
-You only need to create a channel in  rare cases.  Most .NET Core 3 builds should be assigned to the ".NET Core 3 Dev" channel.  However, if you do need to create a new channel:
+You only need to create a channel in  rare cases.  Most .NET Core 5 builds should be assigned to the ".NET Core 5 Dev" channel.  However, if you do need to create a new channel:
```
Comment (jcagme) [inline Documentation/DependencyFlowOnboarding.md]:
```suggestion
You only need to create a channel in rare cases.  Most .NET Core 5 builds should be assigned to the ".NET Core 5 Dev" channel.  However, if you do need to create a new channel:
```

**Comment by dagood** [comment]
This is a breaking change for Core-Setup, for future reference. https://github.com/dotnet/core-setup/issues/8307

**Comment by JohnTortugo** [comment]
This also broke Arcade-validation builds. I think the problem is files outside `eng\common` referencing names that can 'freely' change in post-build templates. I think would be interesting to think about a contract for these names to prevent this kind of problem in the future.. @mmitche

**Pull request merged**